mlrun/datastore/azure_blob.py

```diff
@@ -26,37 +27,58 @@
 # Azure blobs will be represented with the following URL: az://<container name>. The storage account is already
 # pointed to by the connection string, so the user is not expected to specify it in any way.
 
+MAX_CONCURRENCY = 100
+MAX_BLOCKSIZE = 1024 * 1024 * 4
+MAX_SINGLE_PUT_SIZE = 1024 * 1024 * 8  # for BlobServiceClient only.
+
 
 class AzureBlobStore(DataStore):
     using_bucket = True
 
     def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
         super().__init__(parent, name, schema, endpoint, secrets=secrets)
+        self._service_client = None
+        self.max_concurrency = MAX_CONCURRENCY
+        self.max_blocksize = MAX_BLOCKSIZE
+        self.max_single_put_size = MAX_SINGLE_PUT_SIZE
 
     @property
     def filesystem(self):
         """return fsspec file system object, if supported"""
         if self._filesystem:
             return self._filesystem
+        self.create_filesystem_and_client()
+        return self._filesystem
+
+    @property
+    def service_client(self):
+        if not self._filesystem or self._service_client:
+            self.create_filesystem_and_client()
+        return self._service_client
+
+    def create_filesystem_and_client(self):
```

Comment:
Not really sure why these would be created together.

Reply:
I want to make sure they are use the same credentials, even if there is a change (for example, in the env).

Reply:
fixed also in filesystem property.

Reply:
So better save the `storage_options`. There's no need to tie these two things together, so that if one fails, the other becomes unusable as well.